src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py

```diff
@@ -304,3 +334,130 @@ def _info (self):
             'bytes_quota': "infinite" if nsize == 0 else nsize, 'bytes_used': int(usedbytes),
             'bytes_pcent': "undefined" if nsize == 0 else '{0:.2f}'.format((float(usedbytes) / nsize) * 100.0),
             'pool_namespace': pool_namespace, 'features': features, 'state': state}
+
```

Comment:
I think this change would not be required is V2 derives from V1. Implementations are not drastically different for the most part.

Reply:
Applies to [this](https://github.com/ceph/ceph/pull/35647#discussion_r455046581) comment as well.

I decided to move the methods to the baseclass based on the public methods of the base class itself, and let the derived classes override any specifics. Further declaring abstract methods in the base class to help things along for v3, or at least making it more explicit on what derived classes should implement.
NOTE: I should probably add abstract methods for create/open/create_clone as these are missing in the base class, if we continue along this line.

This also helped me rationalize the code more easily, as either defined in base class or in the derived class.

I did not think about deriving v1 from v2, and agree that the move of methods to base class from v1 can be avoided, if we derive v2 from v1.

If it is mandatory or correct that v2 be derived from v1 I can work on that. If it is a matter of choice and either scheme is acceptable, I would like to retain the current one as I find it personally easier to rationalize.

Reply:
> If it is mandatory or correct that v2 be derived from v1 I can work on that. If it is a matter of choice and either scheme is acceptable, I would like to retain the current one as I find it personally easier to rationalize.

It is certainly not mandatory, though, I recommend that we go this route.

Reply:
In lieu of overall code churn due to this patch, moved to deriving V2 from V1, to reduce code movement from V1 into base class.